[ONNX] Add Internal Utils: onnx_proto_utils.py for onnx/onnx-script/onnx_proto

titaiwangms opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #89078
* __->__ #88376
* #86907

Added `onnx_proto_utils.py` for onnx/onnx-script related process. The idea is like jit_utils.py, and to simplify what we have in `torch/onnx/utils.py`

titaiwangms committed:
[ONNX] Add Internal Utils: onnx_proto_utils.py for onnx/onnx-script/onnx_proto

[ghstack-poisoned]

AllenTiTaiWang commented on test/onnx/test_onnxscript_no_runtime.py:
```diff
@@ -92,7 +94,9 @@ def custom_layer_norm(
         y = torch.randn(N, C)
         model_layer_norm = torch.nn.LayerNorm(C)
         layer_norm_onnx = io.BytesIO()
-        torch.onnx.export(model_layer_norm, y, layer_norm_onnx, opset_version=self.opset_version)
```
These lint causing by not doing lint from previous PR. Will fix this.

titaiwangms committed:
Update on "[ONNX] Add Internal Utils: onnx_proto_utils.py for onnx/onnx-script/onnx_proto"




[ghstack-poisoned]

justinchuby commented on torch/onnx/_internal/onnx_proto_utils.py:
```diff
@@ -0,0 +1,116 @@
+        assert len(export_map) == 0
+        with torch.serialization._open_file_like(f, "wb") as opened_file:
+            opened_file.write(model_bytes)
+    elif export_type in [
+        _exporter_states.ExportTypes.ZIP_ARCHIVE,
+        _exporter_states.ExportTypes.COMPRESSED_ZIP_ARCHIVE,
+    ]:
+        compression = (
```
```suggestion
    elif export_type in {
        _exporter_states.ExportTypes.ZIP_ARCHIVE,
        _exporter_states.ExportTypes.COMPRESSED_ZIP_ARCHIVE,
    }:
```

AllenTiTaiWang replied:
Done

justinchuby reviewed (approved):
LGTM🚀

titaiwangms committed:
Update on "[ONNX] Add Internal Utils: onnx_proto_utils.py for onnx/onnx-script/onnx_proto"


Added `onnx_proto_utils.py` for onnx/onnx-script related process. The idea is like jit_utils.py, and to simplify what we have in `torch/onnx/utils.py`


[ghstack-poisoned]

titaiwangms committed:
Update on "[ONNX] Add Internal Utils: onnx_proto_utils.py for onnx/onnx-script/onnx_proto"


Added `onnx_proto_utils.py` for onnx/onnx-script related process. The idea is like jit_utils.py, and to simplify what we have in `torch/onnx/utils.py`


[ghstack-poisoned]

AllenTiTaiWang commented on torch/onnx/_internal/onnx_proto_utils.py:
```diff
@@ -0,0 +1,116 @@
+import torch
+import torch.jit._trace
+import torch.serialization
+from torch.onnx import _constants, _exporter_states, errors  # noqa: F401
```
DOne

AllenTiTaiWang commented on torch/onnx/_internal/onnx_proto_utils.py:
```diff
@@ -0,0 +1,116 @@
+            ) as opened_file:
+                opened_file.write(v)
+    else:
+        raise RuntimeError("Unknown export type")
```
Done

titaiwangms committed:
Update on "[ONNX] Add Internal Utils: onnx_proto_utils.py for onnx/onnx-script/onnx_proto"


Added `onnx_proto_utils.py` for onnx/onnx-script related process. The idea is like jit_utils.py, and to simplify what we have in `torch/onnx/utils.py`


[ghstack-poisoned]

titaiwangms committed:
Update on "[ONNX] Add Internal Utils: onnx_proto_utils.py for onnx/onnx-script/onnx_proto"


Added `onnx_proto_utils.py` for onnx/onnx-script related process. The idea is like jit_utils.py, and to simplify what we have in `torch/onnx/utils.py`


[ghstack-poisoned]

titaiwangms committed:
Update on "[ONNX] Add Internal Utils: onnx_proto_utils.py for onnx/onnx-script/onnx_proto"


Added `onnx_proto_utils.py` for onnx/onnx-script related process. The idea is like jit_utils.py, and to simplify what we have in `torch/onnx/utils.py`


[ghstack-poisoned]

titaiwangms committed:
Update on "[ONNX] Add Internal Utils: onnx_proto_utils.py for onnx/onnx-script/onnx_proto"


Added `onnx_proto_utils.py` for onnx/onnx-script related process. The idea is like jit_utils.py, and to simplify what we have in `torch/onnx/utils.py`


[ghstack-poisoned]

titaiwangms committed:
Update on "[ONNX] Add Internal Utils: onnx_proto_utils.py for onnx/onnx-script/onnx_proto"


Added `onnx_proto_utils.py` for onnx/onnx-script related process. The idea is like jit_utils.py, and to simplify what we have in `torch/onnx/utils.py`


[ghstack-poisoned]

titaiwangms committed:
Update on "[ONNX] Add Internal Utils: onnx_proto_utils.py for onnx/onnx-script/onnx_proto"


Added `onnx_proto_utils.py` for onnx/onnx-script related process. The idea is like jit_utils.py, and to simplify what we have in `torch/onnx/utils.py`


[ghstack-poisoned]

titaiwangms committed:
Update on "[ONNX] Add Internal Utils: onnx_proto_utils.py for onnx/onnx-script/onnx_proto"


Added `onnx_proto_utils.py` for onnx/onnx-script related process. The idea is like jit_utils.py, and to simplify what we have in `torch/onnx/utils.py`


[ghstack-poisoned]

AllenTiTaiWang commented:
@pytorchbot rebase

titaiwangms committed:
Update on "[ONNX] Add Internal Utils: onnx_proto_utils.py for onnx/onnx-script/onnx_proto"


Added `onnx_proto_utils.py` for onnx/onnx-script related process. The idea is like jit_utils.py, and to simplify what we have in `torch/onnx/utils.py`


[ghstack-poisoned]

titaiwangms committed:
Update on "[ONNX] Add Internal Utils: onnx_proto_utils.py for onnx/onnx-script/onnx_proto"


Added `onnx_proto_utils.py` for onnx/onnx-script related process. The idea is like jit_utils.py, and to simplify what we have in `torch/onnx/utils.py`


[ghstack-poisoned]

AllenTiTaiWang commented:
@pytorchbot merge -g
